### PR TITLE
Throw warning on changing val_loss

### LINKED ISSUE #2868: Throw warning for changing val_loss
Add warning to user that when changing val_loss to another keyword it will break checkpointing, early stopping, and other features relying on it.

### PR DESCRIPTION
**Pull request by shivin7**
## What does this PR do?

Fixes #2868 

# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

Please note : I'm a little new with the tests so any comments are really helpful. Thanks!

### CODE REVIEW

**Comment by shivin7** [comment]
Right now I'm throwing the same warning when the output of `validation_step` is a scalar or `None`. Is that fine?

File: pytorch_lightning/trainer/evaluation_loop.py [hunk 1]
```diff
@@ -343,6 +343,11 @@ def _evaluate(
                     m = 'only EvalResults or dicts are allowed from validation_step'
                     raise MisconfigurationException(m)
 
+                # throw warning if key other than 'val_loss' is used in output
+                if not isinstance(output, EvalResult) or 'val_loss' not in output.keys():
```
Comment (williamFalcon) [inline pytorch_lightning/trainer/evaluation_loop.py:347]: that’s not what we meant. 

the warning js when the word monitor changes in a callback AND you are using a result obj

Comment (shivin7) [reply pytorch_lightning/trainer/evaluation_loop.py:347]: Oh I see. 
So something like when `EarlyStopping(monitor='not_val_loss')` AND using a `Result` object?

Comment (williamFalcon) [reply pytorch_lightning/trainer/evaluation_loop.py:347]: correct

**Comment by williamFalcon** [comment]
this is time sensitive though, so i can fix it

**Pull request closed without merging**